pkg/orchestration/wiring/wiringplugin/shapes.go

```diff
@@ -39,3 +46,30 @@ type ShapeMeta struct {
 type BindableShapeStruct struct {
 	ServiceInstanceName ProtoReference `json:"serviceInstanceName"`
 }
+
+// CopyShape copies source shape into the target shape.
+// The main purpose of this function is to convert between typed and UnstructuredShape representations
+// of the same shape.
+func CopyShape(source, target Shape) error {
+	st := reflect.TypeOf(source)
+	tt := reflect.TypeOf(target)
+	switch {
+	case st == tt:
+		// Same type already, just deep copy
+		sv := reflect.ValueOf(source)
+		tv := reflect.ValueOf(target)
+		method := sv.MethodByName("DeepCopyInto")
+		method.Call([]reflect.Value{tv})
+	default:
+		// Convert by round-tripping to and from JSON
+		data, err := json.Marshal(source)
```

Comment:
Perhaps we should look for a specific converter first?

Reply:
What do you mean?

Reply:
It'd be nice if there were a way to define a map(from, to) function instead of falling back to serializing and then deserializing to convert any different shapes.

Reply:
Yes, that is how scheme `k8s.io/apimachinery/pkg/runtime.Scheme` works. I thought about it but I think it would be overkill. At least for now we don't need such mechanism, and we can build it later. I've also thought about using `k8s.io/apimachinery/pkg/runtime.DefaultUnstructuredConverter` somehow but there are some potential issues with using it (name of the field in struct not matching the name of the field in JSON) so I stopped looking into it.